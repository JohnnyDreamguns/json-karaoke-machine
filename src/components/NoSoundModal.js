import React, { useState } from 'react';
import Modal from 'react-modal';

//Modal.setAppElement('#root');

const NoSoundModal = () => {
  const [visible, setVisible] = useState(false);

  return (
    <div>
      <button
        id="btn-no-sound"
        className="btn-no-sound"
        onClick={() => setVisible(true)}
      >
        No Sound?
      </button>
      <Modal
        isOpen={visible}
        contentLabel="Example Modal"
        className="no-sound-modal"
      >
        <h2>Hello</h2>
        <button id="btn-close" onClick={() => setVisible(false)}>
          X
        </button>
        <p>
          If you can't hear anything when it looks like it should be playing
          then please check that your volume is up.
        </p>

        <p>
          If using a mobile please ensure that the silent / mute switch is off.
        </p>
        <p>
          If you still can't hear anything please raise an issue on the{' '}
          <a href="https://github.com/JohnnyDreamguns/json-karaoke-machine">
            GitHub page
          </a>
          , including your OS, browser and version.
        </p>
        <p>Thanks!</p>
      </Modal>
      <input
        type="hidden"
        id="isVisible"
        value={visible ? 'visible' : 'hidden'}
      ></input>
    </div>
  );
};

export default NoSoundModal;
