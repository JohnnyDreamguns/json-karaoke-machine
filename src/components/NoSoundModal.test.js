import 'babel-polyfill';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NoSoundModal from './NoSoundModal';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('NoSoundModal', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NoSoundModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should set visible to true when no-sound button is pressed', () => {
    let wrapper = Enzyme.shallow(<NoSoundModal />);
    wrapper.find('#btn-no-sound').prop('onClick')();
    expect(wrapper.find('#isVisible').props().value).toBe('visible');
  });

  it('should set visible to false when btn-close button is pressed', () => {
    let wrapper = Enzyme.shallow(<NoSoundModal />);
    wrapper.find('#btn-no-sound').prop('onClick')();
    wrapper.find('#btn-close').prop('onClick')();
    expect(wrapper.find('#isVisible').props().value).toBe('hidden');
  });
});
