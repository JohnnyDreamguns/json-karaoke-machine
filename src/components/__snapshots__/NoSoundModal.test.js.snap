// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSoundModal renders correctly 1`] = `
<div>
  <button
    className="btn-no-sound"
    id="btn-no-sound"
    onClick={[Function]}
  >
    No Sound?
  </button>
  <input
    id="isVisible"
    type="hidden"
    value="hidden"
  />
</div>
`;
